fix: match pattern file with Meshery Design schema

weilirs opened the pull request:
**Description**
https://github.com/meshery/meshery-cilium/pull/173#issuecomment-2564600196

This PR fixes [this issue](https://github.com/meshery/meshery/issues/13024)

**Notes for Reviewers**
Followed the instructions from Pranav, exported a design file with a cilium component from Kanvas and replaced the content of it with the existing `deploy.yaml`
https://layer5io.slack.com/archives/C017NA80S66/p1736126652434049?thread_ts=1736126645.139429&cid=C017NA80S66

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

leecalcote reviewed (changes requested):
@weilirs how was this design created? 

Change the title from "Untitled Design".

When I import this design into a Meshery Server, its validation fails with a missing property.

leecalcote commented:
As you work through this, please confirm that the design is deployable.

weilirs commented:
> @weilirs how was this design created?
> 
> Change the title from "Untitled Design".
> 
> When I import this design into a Meshery Server, its validation fails with a missing property.

I ran meshery-cilium locally and ran the Meshery server, navigated to the Kanvas and dragged and dropped the "Cilium Clusterwide Envoy Config" and exported it as "Design File" and I replaced the content of `deploy.yaml` with the content of the exported Design File. I now exported a new one and named it as "pattern file" and I tried importing it and succeeded.
![image](https://github.com/user-attachments/assets/f0a07680-0a2d-4581-86a9-891144f7b277)

leecalcote commented:
@weilirs, good progress. Two other aspects in need of confirmation:

1. That the component used in the new design comes from the adapter.
2. That the design created deploys.
3. That the design deployment creates a Cilium service mesh.

sangramrath commented:
> @weilirs, good progress. Two other aspects in need of confirmation:
> 
>     1. That the component used in the new design comes from the adapter.
> 
>     2. That the design created deploys.
> 
>     3. That the design deployment creates a Cilium service mesh.

weilirs commented:
> @weilirs any updates on the changes requested?

I encountered a problem confirming the details Lee mentioned. I asked in slack and the forum but have not heard back.
https://discuss.layer5.io/t/error-when-deploying-a-design/6703
